### ISSUE TITLE
Angular 17 build process raises warnings for link to CSS file within index.html

### ISSUE THREAD

**Issue by caboodal** [angular/angular-cli#26525]
### Command

build

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

16.0.3

### Description

When including a a link within the <head> tag of the index.html the build process raises a warning.

### Minimal Reproduction

To replicate the problem follow these steps: -

Create a new Angular app with the following options: -

1. SSR: No
2. SASS styles

Create a css file within the assets folder.

Add the following <link> tag into the <head> tag of the index.html file: -

````
<link rel="stylesheet" type="css" href="assets/test.css" />
````

From a command prompt run `ng build`

### Exception or Error

```text
Building...▲ [WARNING] Unable to locate stylesheet: C:\assets\test.css
```

### Your Environment

```text
Angular CLI: 17.0.3
Node: 20.9.0
Package Manager: npm 9.8.1
OS: win32 x64

Angular: 17.0.4
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.3
@angular-devkit/build-angular   17.0.3
@angular-devkit/core            17.0.3
@angular-devkit/schematics      17.0.3
@angular/cli                    17.0.3
@schematics/angular             17.0.3
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2
```

**Comment by alan-agius4** [comment]
How is `inter.css` referenced?

**Comment by caboodal** [comment]
sorry, as always as soon as take the time to post an issue I discover the problem/solution. Firstly I mistyped the description above it should have been test.css not inter.css!

I then noticed in my angular.json in my working app it had this in the assets node: -

````
{
  "glob": "_redirects",
  "input": "src",
  "output": "/"
}
````

and then there was a _redirects file contain the following: -

````
/*  /index.html  200
````

This clearly prevents the warning by returning a 200 response. Why I have no idea I'm looking at some legacy code.
